lib/main.coffee

```diff
@@ -0,0 +1,51 @@
+{CompositeDisposable} = require 'atom'
+
+module.exports =
+  config:
+    shellcheckExecutablePath:
+      type: 'string'
+      title: 'Shellcheck Executable Path'
+      default: 'shellcheck' # Let OS's $PATH handle the rest
+    enableNotice:
+      type: 'boolean'
+      title: 'Enable Notice Messages'
+      default: false
+
+  activate: ->
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.config.observe \
+     'linter-shellcheck.shellcheckExecutablePath',
+      (executablePath) =>
+        @executablePath = executablePath
+    @subscriptions.add atom.config.observe 'linter-shellcheck.enableNotice',
+      (enableNotice) =>
+        @enableNotice = enableNotice
+
+  deactivate: ->
+    @subscriptions.dispose()
+
+  provideLinter: ->
+    helpers = require('atom-linter')
+    provider =
+      grammarScopes: ['source.shell']
+      scope: 'file'
+      lintOnFly: true
+      lint: (textEditor) =>
+        filePath =  textEditor.getPath()
+        showAll = @enableNotice
+        parameters = ['-f', 'gcc', filePath ]
+        return helpers.exec(@executablePath, parameters).then (output) ->
+          regex = /.+?:(\d+):(\d+):\s(\w+?):\s(.+)/g
```

Comment:
Strange, this works for me. Probably you cant see this in UI because its a "note" message. I guess you are running an older version than me, Unused variables are shown as warning in the actual version.